### PR TITLE
HDDS-9745. Change IllegalStateException to IOException in SnapshotChainManager

### PR DESCRIPTION
**Pull request by hemantk-12**
## What changes were proposed in this pull request?
As fix of HDDS-9199, exceptions in snapshot chain creation were changed IllegalStateException. We have seen couple of issues with this as in HDDS-9685 and HDDS-9745.
This change is to change unchecked `IllegalStateException` to checked `IOException` so that OM state machine can handle the exception properly and doesn't crash the system.

## What is the link to the Apache JIRA
[HDDS-9745](https://issues.apache.org/jira/browse/HDDS-9745)

## How was this patch tested?
Updated unit and integration tests.

### CODE REVIEW

File: hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/snapshot/OMSnapshotCreateRequest.java [hunk 1]
```diff
@@ -300,7 +300,7 @@ private void removeSnapshotInfoFromSnapshotChainManager(
   ) {
     try {
       snapshotChainManager.deleteSnapshot(info);
-    } catch (IllegalStateException exception) {
+    } catch (IOException exception) {
```
Comment (aswinshakil) [inline hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/snapshot/OMSnapshotCreateRequest.java:303]: Do you want to remove this now that ISE is changed to IOException? 
https://github.com/apache/ozone/blob/f8c42fa86724d5de0ee6511b2af8fb444c4611f4/hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/snapshot/OMSnapshotCreateRequest.java#L194

Comment (hemantk-12) [reply hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/snapshot/OMSnapshotCreateRequest.java:303]: Ah, I missed it. Removed.

Thanks for the catch.

**Review by aswinshakil — APPROVED** [review]
LGTM+1. Pending CI. Thanks for the fix @hemantk-12

**Comment by hemantk-12** [comment]
Thanks for the review @aswinshakil

**Pull request merged**